src/components/bootcamp/BootcampGrid.tsx

```diff
@@ -0,0 +1,290 @@
+    setOpenRowsDel(false)
+  }
+
+  const loadInternInfo = async (id: string) => {
+    try {
+      await infoMutation.mutateAsync(id)
+    } catch (error) {
+      console.error(error)
+    }
+  }
+
+  const closeInfoDialog = () => {
+    setOpenInfo(false)
+  }
+
+  const infoMutation = useMutation<AxiosResponse<BootcampResponse>, AxiosError<ApiErrors>, string>({
+    mutationFn: getBootcampIntern,
+    onError: () => AlertStore.show(t('bootcamp:alerts.load-intern.error'), 'error'),
+    onSuccess: ({ data }) => {
+      setIntern(data)
+      setOpenInfo(true)
+    },
+  })
+
+  const delMutation = useMutation<AxiosResponse<BootcampResponse>, AxiosError<ApiErrors>, string>({
+    mutationFn: deleteBootcampIntern,
+    onError: () => AlertStore.show(t('bootcamp:alerts.delete-row.error'), 'error'),
+    onSuccess: () => {
+      return
+    },
+  })
+
+  const deleteRow = async () => {
+    try {
+      closeDeleteRowDialog()
+      delMutation.mutateAsync(selectedId)
+      AlertStore.show(t('bootcamp:alerts.delete-row.success'), 'success')
+      router.push(routes.bootcamp.index)
+    } catch (err) {
+      console.log(err)
+    }
+  }
+
+  const selectMultipleRows = (ids: GridSelectionModel) => {
+    setMultipleSelectedIds(ids.map((id) => id.toString()))
+  }
+
+  const deleteRows = async () => {
+    try {
+      closeDeleteRowsDialog()
+      await Promise.all(multipleSelectedIds.map((id) => delMutation.mutateAsync(id)))
+      AlertStore.show(t('bootcamp:alerts.delete-rows.success'), 'success')
+      setMultipleSelectedIds([])
+      router.push(routes.bootcamp.index)
+    } catch (err) {
+      AlertStore.show(t('bootcamp:alerts.delete-rows.error'), 'error')
+    }
+  }
+
+  type ActionsProps = {
+    id: string
+  }
+
+  const ActionsButtons = ({ id }: ActionsProps) => {
+    return (
+      <>
+        <IconButton onClick={() => loadInternInfo(id)}>
+          <PreviewIcon />
+        </IconButton>
+        <Link href={routes.bootcamp.view(id)} passHref>
+          <IconButton>
+            <EditIcon />
+          </IconButton>
+        </Link>
+        <IconButton onClick={() => openDeleteRowDialog(id)}>
+          <DeleteIcon />
+        </IconButton>
+      </>
+    )
+  }
+
+  const columns: GridColumns = [
+    { field: 'id', headerName: 'ID', hide: true },
+    {
+      field: 'firstName',
+      headerName: t('auth:fields.first-name'),
+      valueGetter: (p) => p.row.firstName,
+      width: 200,
+    },
+    {
+      field: 'lastName',
+      headerName: t('auth:fields.last-name'),
+      valueGetter: (p) => p.row.lastName,
+      width: 200,
+    },
+    {
+      field: 'phone',
+      headerName: t('auth:fields.phone'),
+      valueGetter: (p) => p.row.phone,
+      width: 200,
+    },
+    {
+      field: 'actions',
+      headerName: '',
+      renderCell: (p) => <ActionsButtons id={p.row.id} />,
+      width: 150,
+      type: 'actions',
+    },
+  ]
+
+  const DeleteRowDialog = () => (
```

Comment:
You can extract all these inline components in standalone components to achieve better reusability.


Now only `BootcampGrid` is able to use `DeleteRowDialog` but in the future multiple grids may use such component